src/app/core/layers/providers/wfsprovider.js

```diff
@@ -12,137 +12,137 @@ inherit(WFSDataProvider, DataProvider);
 const proto = WFSDataProvider.prototype;
 
 proto.getData = function() {
-  const d = $.Deferred();
-  return d.promise();
+  return new Promise((resolve, reject) => {})
 };
 
 // query method
 proto.query = function(options={}, params = {}) {
   const {reproject=false, feature_count=10, filter} = options;
   params.MAXFEATURES = feature_count;
-  const d = $.Deferred();
-  const {layers=[this._layer]} = options;
+  return new Promise((resolve, reject) => {
+    const {layers=[this._layer]} = options;
 
-  const projections = {
-    map: this._layer.getMapProjection(),
-    layer: reproject ? this._layer.getProjection(): null
-  };
+    const projections = {
+      map: this._layer.getMapProjection(),
+      layer: reproject ? this._layer.getProjection(): null
+    };
 
-  const timeoutKey = this.getQueryResponseTimeoutKey({
-    layers,
-    resolve: d.resolve,
-    query:{}
-  });
+    const timeoutKey = this.getQueryResponseTimeoutKey({
+      layers,
+      resolve: resolve,
+      query:{}
+    });
 
-  this._doRequest(filter, params, layers, reproject)
-    .then(response => {
-      const featuresForLayers = this.handleQueryResponseFromServer(response, projections, layers, wms=false);
-      featuresForLayers.forEach(featuresForLayer => {
-        const {features=[]} = featuresForLayer;
-        //sanitize in case of nil:true
-        features.forEach(feature => {
-          Object.entries(feature.getProperties()).forEach(([attribute, value])=>{
-            if (toRawType(value) === 'Object' && value['xsi:nil'])feature.set(attribute, 'NULL');
+    this._doRequest(filter, params, layers, reproject)
+      .then(response => {
+        const featuresForLayers = this.handleQueryResponseFromServer(response, projections, layers, wms=false);
+        featuresForLayers.forEach(featuresForLayer => {
+          const {features=[]} = featuresForLayer;
+          //sanitize in case of nil:true
+          features.forEach(feature => {
+            Object.entries(feature.getProperties()).forEach(([attribute, value])=>{
+              if (toRawType(value) === 'Object' && value['xsi:nil'])feature.set(attribute, 'NULL');
+            })
           })
-        })
-      });
-      d.resolve({
-        data: featuresForLayers
+        });
+        resolve({
+          data: featuresForLayers
+        });
+      })
+      .catch(e => reject(e))
+      .finally(()=> {
+        clearTimeout(timeoutKey)
       });
-    })
-    .fail(e => d.reject(e))
-    .always(()=> {
-      clearTimeout(timeoutKey)
-    });
-  return d.promise();
+  })
 };
 
 proto._post = function(url, params) {
   url = url.match(/\/$/) ? url : `${url}/`;
-  const d = $.Deferred();
-  $.post(url, params)
-    .then(response => d.resolve(response))
-    .fail(err => d.reject(err));
-  return d.promise();
+  return new Promise((resolve, reject) => {
+    XHR.post({
+      url,
+      data:params
+    }).then(response => resolve(response))
+      .catch(err => reject(err));
```

Comment:
double check, written that way wouldn't the whole surrounding promise above be superfluous?

```js
// return new Promise((resolve, reject) => {
    XHR.post({ ... })
//    .then(response => resolve(response))
//    .catch(err => reject(err));
// });
```